### PR TITLE
Add service orchestration guide

### PR DESCRIPTION
**Pull request by anuruddhal**
## Purpose
- Add service orchestration guide

### CODE REVIEW

File: en/docs/learn/message-routing.md [hunk 1]
```diff
@@ -245,4 +246,5 @@ Follow the steps below to implement the message routing service.
     "confirmed": false,
     "appointmentDate": "2023-10-02"
```
Comment (anupama-pathirage) [inline en/docs/learn/message-routing.md:247]: This is not properly indented

File: en/docs/learn/service-orchestration.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+4. Add another connector for the payment backend service with the URL `http://localhost:9090/healthcare/payments` and the name `paymentEP`, and config as `{}`.    
+<a href="{{base_path}}/assets/img/service-orchestration/connectors.png"><img src="{{base_path}}/assets/img/service-orchestration/connectors.png" alt="Connectors" width="70%"></a>
+
+???+ info "HTTP Connector"
+    To learn more about HTTP client, see [Ballerina HTTP Client](https://ballerina.io/learn/by-example/http-client-send-request-receive-response/).
+    See supported client configurations in the [HTTP Client Configurations](https://central.ballerina.io/ballerina/http/2.12.2#ClientConfiguration).
+
+### Step 5: Design the resource
+1. The service will be generated with a default resource named `greeting`. Click on the service to view and update the resource details.
+2. Click on three dots appear in front of the `greeting` resource and select edit to view and update the resource details.
```
Comment (anupama-pathirage) [inline en/docs/learn/service-orchestration.md:139]:
```suggestion
2. Click on three dots that appear in front of the `greeting` resource and select edit to view and update the resource details.
```

File: en/docs/learn/service-orchestration.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+          }
+         doctor: reservation.doctor,
+         hospital: reservation.hospital,
+         appointment_date: reservation.appointment_date
+        }
+    ```
+5. Click on the **`Save`** button to add the variable.   
+<a href="{{base_path}}/assets/img/service-orchestration/variable.png"><img src="{{base_path}}/assets/img/service-orchestration/variable.png" alt="Variable" width="70%"></a>
+6. Click ➕ sign and select **`hospitalServicesEP`** connector from the node panel.
+7. Select **`post`** from the dropdown. Then fill the required fields with values given below and click **`Save`**.
```
Comment (anupama-pathirage) [inline en/docs/learn/service-orchestration.md:170]:
```suggestion
7. Select **`post`** from the dropdown. Then, fill the required fields with values given below and click **`Save`**.
```

File: en/docs/learn/service-orchestration.md [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        doctor: appointment.doctor.toJson(),
+        patient: appointment.patient.toJson(),
+        fee: actualFee,
+        confirmed: false,
+        card_number: reservation.patient.cardNo
+    }
+    ```
+   <a href="{{base_path}}/assets/img/service-orchestration/payment-request.png"><img src="{{base_path}}/assets/img/service-orchestration/payment-request.png" alt="Payment Request" width="70%"></a>  
+13. Let's add another connector action to make the payment. Click on the ➕ sign and select **`paymentEP`** connector from the node panel.   
+14. Select **`post`** from the dropdown. Then fill the required fields with values given below and click **`Save`**.
```
Comment (anupama-pathirage) [inline en/docs/learn/service-orchestration.md:206]: ditto

File: en/docs/learn/message-routing.md [hunk 5]
```diff
@@ -153,16 +153,17 @@ Follow the steps below to implement the message routing service.
...
  conditions:
     * grandOak -> `reservation.hospital_id == "grandoak"`
     * clemency -> `reservation.hospital_id == "clemency"`
     * pineValley -> `reservation.hospital_id == "pinevalley"`
    <a href="{{base_path}}/assets/img/message-routing/add-if.png"><img src="{{base_path}}/assets/img/message-routing/add-if.png"" alt="Add If" width="70%"></a>
-7. Select the `grandOakEP` condition true path ➕ sign and select **`grandOakEP`** connector from the node panel.
+8. Select the `grandOakEP` condition true path ➕ sign and select **`grandOakEP`** connector from the node panel.
 <a href="{{base_path}}/assets/img/message-routing/add-connector-action.png"><img src="{{base_path}}/assets/img/message-routing/add-connector-action.png" alt="Add Connector Action" width="70%"></a>
-8. Select **`Post`** from the dropdown. Then fill the required fields with values given below and click **`Save`**.
+9. Select **`post`** from the dropdown. Then fill the required fields with values given below and click **`Save`**.
```
Comment (anupama-pathirage) [inline en/docs/learn/message-routing.md:166]:
```suggestion
9. Select **`post`** from the dropdown. Then, fill in the required fields with the values given below and click **`Save`**.
```

File: en/docs/learn/service-orchestration.md (hunk 4 above)
Comment (anupama-pathirage) [inline en/docs/learn/service-orchestration.md:206]:
```suggestion
14. Select **`post`** from the dropdown. Then, fill in the required fields with the values given below and click **`Save`**.
```

File: en/docs/learn/service-orchestration.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+10. Select **`get`** from the dropdown. Then fill the required fields with values given below and click **`Save`**.
+
+   |Field| Value                                                            |
+         |---|------------------------------------------------------------------|
+   |Variable Name| `fee`                                                            |
+   |Variable Type| `Fee`                                                            |
+   |Resource Path| `/[reservation.hospital_id]/categories/appointments/[appointmentNumber]/fee` |
+
+   <a href="{{base_path}}/assets/img/service-orchestration/fee.png"><img src="{{base_path}}/assets/img/service-orchestration/fee.png" alt="Hospital Service Request" width="70%"></a>  
+11. Click on the ➕ sign and select **`Variable`** from the node panel. Add a variable named `actualFee` with the type **`dobule`** and expression `decimal:fromString(fee.actualFee)`.
```
Comment (anupama-pathirage) [inline en/docs/learn/service-orchestration.md:192]:
```suggestion
11. Click on the ➕ sign and select **`Variable`** from the node panel. Add a variable named `actualFee` with the type **`double`** and expression `decimal:fromString(fee.actualFee)`.
```

**Review by sm1990 — DISMISSED** [review]

**Review by sm1990 — APPROVED** [review]

**Pull request merged**